src/tt_store.erl

```diff
@@ -1,9 +1,94 @@
 -module(tt_store).
 
--export([find_by_time/2]).
+-behaviour(gen_server).
+-define(SERVER, ?MODULE).
 
--record(talk, {title :: string(), start_time :: calendar:datetime(), end_time :: calendar:datetime(), location :: string()}).
+-include_lib("stdlib/include/ms_transform.hrl").
+%% API
+-export([start_link/0, add/4, find_by_time/2, list/0, find_by_name/1]).
 
--spec(find_by_time(calendar:datetime(), calendar:datetime()) -> [#talk{}]).
-find_by_time(_StartTime, _EndTime) ->
-  [#talk{title = "Title", start_time = calendar:local_time(), end_time = calendar:local_time(), location =  "Location"}].
+%% Supervisor callbacks
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2,
+         terminate/2, code_change/3]).
+
+%% DB record
+-record(talk, {title :: string(), startTime :: {Hour, Minute, Second}, endTime :: {Hour, Minute, Second}, location :: any()}).
+
+%%====================================================================
+%% API functions
+%%====================================================================
+
+start_link() ->     
+    gen_server:start_link({local, ?SERVER},?MODULE, [], []).
+
+add(Title, StartTime, EndTime, Where) -> 
+    gen_server:cast(?SERVER, {add, {Title, StartTime, EndTime, Where}}),
+    ok.
+
+find_by_time(StartTime, EndTime) -> 
```

Comment:
`find_by_time/2` is incorrect in 2 aspects:
1. It relies only on `calendar:time()` instead of `calendar:datetime()`. I.e. it should lookup the entries based not only on HH:MM:SS but on the date too.
2. It is expected to return talks within a **time range** - not matching the exact start/end times.

Here is a test that reveals that misconception:
```erlang
test4(_) ->
    tt_store:start_link(),
    tt_store:add(A = "FirstTalk", {10,0,0}, {11,0,0}, "Somewhere"),
    tt_store:add(B = "SecondTalk", {10,30,0}, {11,30,0}, "Somewhere"),
    tt_store:add("ThirdTalk", {12,0,0}, {13,00,0}, "Somewhere"),
    ?assertMatch([{A, '_', '_', '_'}, {B, '_', '_', '_'}],
                 tt_store:find_by_time({10,0,0}, {12,0,0})).
```

Note: this test is based only on the HH:MM:SS which is of course incorrect.